slurm-abstract/slurmd.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+NODE=${SLURM_NODE_NAME:-slurm-worker-0}
+
+mkdir -p /run/slurm/
+chown -R slurm:root /run/slurm
+mkdir -p /var/log/slurm
+chown -R slurm:root /var/log/slurm
+
+# intentionally run as root
+# see https://slurm.schedmd.com/quickstart_admin.html
+exec /usr/sbin/slurmd -D -N $NODE > /var/log/slurm/slurmd-${NODE}.out.log 2> /var/log/slurm/slurmd-${NODE}.err.log
```

Comment:
Typically the primary process in a docker container runs in the foreground and outputs its logs to stdout. There's a lot of stuff going on here that I don't understand, so maybe that's impossible. Just wanted to make sure that we don't lose touch with docker in all the process orchestration.

Reply:
Totally. Way up the chain of docker image hierarchy, this is based on `phusion/baseimage`.
https://github.com/phusion/baseimage-docker

Reply:
Aha! This image already fills in a lot of gaps for me. Thanks!

Reply:
My thoughts exactly. Now that I get what they're doing with `my_init` and that they seem to have a battle tested solution, I'm pretty ok with it. Sometimes daemons travel in herds. I'm going to keep this in my back pocket for sure. 💯